kubelet panic

**version:**
Kubernetes v1.7.0-alpha.0.1164+e823af8a873bda   (compiled by the newest code)

**command:**
./kubelet  --hostname-override=10.62.62.106 --api-servers==http://10.62.62.104:8080/  --logtostderr=true --log-dir=/tmp &

**logs:**
```
[root@admin k8s]# ./kubelet_power_on.sh 
[root@admin k8s]# Flag --api-servers has been deprecated, Use --kubeconfig instead. Will be removed in a future version.
I0315 09:00:12.559527   19231 feature_gate.go:144] feature gates: map[]
W0315 09:00:12.560152   19231 server.go:715] Could not load kubeconfig file /var/lib/kubelet/kubeconfig: stat /var/lib/kubelet/kubeconfig: no such file or directory. Using default client config instead.
W0315 09:00:12.560230   19231 server.go:450] New kubeClient from clientConfig error: host must be a URL or a host:port pair: "=http://10.62.62.104:8080/"
W0315 09:00:12.560270   19231 server.go:454] New kubeClient from clientConfig error: host must be a URL or a host:port pair: "=http://10.62.62.104:8080/"
W0315 09:00:12.560335   19231 server.go:462] Failed to create API Server client: host must be a URL or a host:port pair: "=http://10.62.62.104:8080/"
I0315 09:00:12.560944   19231 docker.go:364] Connecting to docker on unix:///var/run/docker.sock
I0315 09:00:12.560966   19231 docker.go:384] Start docker client with request timeout=2m0s
W0315 09:00:12.564313   19231 cni.go:163] Unable to update cni config: No networks found in /etc/cni/net.d
I0315 09:00:12.575471   19231 manager.go:143] cAdvisor running in container: "/user.slice"
W0315 09:00:12.598945   19231 manager.go:151] unable to connect to Rkt api service: rkt: cannot tcp Dial rkt api service: dial tcp [::1]:15441: getsockopt: connection refused
I0315 09:00:12.623030   19231 fs.go:117] Filesystem partitions: map[/dev/mapper/centos_admin-root:{mountpoint:/var/lib/docker/overlay major:253 minor:1 fsType:xfs blockSize:0} /dev/vda1:{mountpoint:/boot major:252 minor:1 fsType:xfs blockSize:0}]
I0315 09:00:12.631146   19231 manager.go:198] Machine: {NumCores:4 CpuFrequency:1899999 MemoryCapacity:3975593984 MachineID:1ff9fa702e864c49b0a6bb5d3952df3e SystemUUID:7036B7C6-09D0-C642-981F-50D8B463D786 BootID:10acdeef-06f6-457b-a8cc-f84bc02c9f0a Filesystems:[{Device:/dev/vda1 Capacity:520794112 Type:vfs Inodes:512000 HasInodes:true} {Device:/dev/mapper/centos_admin-root Capacity:50940104704 Type:vfs Inodes:49770496 HasInodes:true}] DiskMap:map[253:0:{Name:dm-0 Major:253 Minor:0 Size:2147483648 Scheduler:none} 253:1:{Name:dm-1 Major:253 Minor:1 Size:50964987904 Scheduler:none} 252:0:{Name:vda Major:252 Minor:0 Size:64424509440 Scheduler:none}] NetworkDevices:[{Name:eth0 MacAddress:fa:16:3e:77:ae:40 Speed:0 Mtu:1500}] Topology:[{Id:0 Memory:4294434816 Cores:[{Id:0 Threads:[0 1] Caches:[{Size:4194304 Type:Unified Level:2}]}] Caches:[]} {Id:1 Memory:0 Cores:[{Id:0 Threads:[2 3] Caches:[{Size:4194304 Type:Unified Level:2}]}] Caches:[]}] CloudProvider:Unknown InstanceType:Unknown InstanceID:None}
I0315 09:00:12.631845   19231 manager.go:204] Version: {KernelVersion:3.10.0-229.el7.x86_64 ContainerOsVersion:CentOS Linux 7 (Core) DockerVersion:17.03.0-ce CadvisorVersion: CadvisorRevision:}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x51d385]

goroutine 1 [running]:
panic(0x2d90de0, 0xc4200120a0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
k8s.io/kubernetes/vendor/k8s.io/client-go/kubernetes.(*Clientset).Events(0x0, 0x0, 0x0, 0x0, 0x0)
	<autogenerated>:25 +0x5
k8s.io/kubernetes/cmd/kubelet/app.makeEventRecorder(0xc4206cec00, 0xc4203e57c0, 0x7fff649d27d8, 0xc)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubelet/app/server.go:348 +0x1ba
k8s.io/kubernetes/cmd/kubelet/app.run(0xc4206cec00, 0xc4203e57c0, 0xc420763ec0, 0x456eb0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubelet/app/server.go:505 +0x2c2
k8s.io/kubernetes/cmd/kubelet/app.Run(0xc4206cec00, 0x0, 0x22, 0xc420418be0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubelet/app/server.go:285 +0x39
main.main()
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubelet/kubelet.go:48 +0x92
```

--- a/cmd/kubelet/app/server.go
+++ b/cmd/kubelet/app/server.go
@@ -159,6 +159,7 @@ func UnsecuredDependencies(s *options.KubeletServer) (*kubelet.Dependencies, err
 		DockerClient:       dockerClient,
 		KubeClient:         nil,
 		ExternalKubeClient: nil,
+		EventClient:        nil,
 		Mounter:            mounter,
 		NetworkPlugins:     ProbeNetworkPlugins(s.NetworkPluginDir, s.CNIConfDir, s.CNIBinDir),
 		OOMAdjuster:        oom.NewOOMAdjuster(),
@@ -502,7 +503,9 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.Dependencies) (err error) {
 
 		kubeDeps.KubeClient = kubeClient
 		kubeDeps.ExternalKubeClient = externalKubeClient
-		kubeDeps.EventClient = eventClient
+		if eventClient != nil {
+			kubeDeps.EventClient = eventClient
+		}
 	}
 
 	if kubeDeps.Auth == nil {

